jetty-client/src/main/java/org/eclipse/jetty/client/AuthenticationProtocolHandler.java

```diff
@@ -241,8 +241,8 @@ private void forwardFailureComplete(HttpRequest request, Throwable requestFailur
                 if (matcher.matches())
                 {
                     String type = matcher.group(1);
-                    String realm = matcher.group(2);
-                    String params = matcher.group(3);
+                    String realm = matcher.group(3);
+                    String params = matcher.group(2) + matcher.group(4);
```

Comment:
With the example above, `params` becomes:
```
qop="auth", blah, nonce="1523430383"
```

In the good case, it becomes:
```
qop="auth", , nonce="1523430383"
```

I'm not sure the rest of the code is able to parse the double comma properly.

Reply:
i checked that and the rest of the code seems to be fine handling the double comma
in fact if you add this line in after setting params it still passes all the tests
`params = ",  , , , ,  , ," + params + ",  , , , ,, , ,  ,";`